manim/mobject/geometry/polygram.py

```diff
@@ -121,7 +122,7 @@ def get_vertices(self) -> InternalPoint3D_Array:
         """
         return self.get_start_anchors()
 
-    def get_vertex_groups(self) -> InternalPoint3D_Array:
+    def get_vertex_groups(self) -> npt.NDArray[ManimFloat]:
```

Comment:
good catch :)

Reply:
Thanks!

I noticed that the current implementation of `.get_vertex_groups()` is faulty: it returns a 3D NumPy array where each of its elements is a vertex group corresponding to a `Point2D_Array`, but these groups shouldn't necessarily all have the same length and this method would crash if that's not the case. It should return, instead, a `list[Point2D_Array]`, but I decided to do this in a subsequent PR, because the change is a bit more complex than it seems at first.